backend/entityservice/cache/encodings.py

```diff
@@ -0,0 +1,56 @@
+import pickle
+import structlog
+
+from entityservice.object_store import connect_to_object_store
+from entityservice.serialization import binary_unpack_filters
+from entityservice.cache.connection import connect_to_redis
+from entityservice.database import DBConn, get_filter_metadata
+from entityservice.settings import Config as config
+
+logger = structlog.get_logger()
+
+
+def set_deserialized_filter(dp_id, python_filters):
+    if len(python_filters) <= config.MAX_CACHE_SIZE:
+        logger.debug("Pickling filters and storing in redis")
+        key = 'clk-pkl-{}'.format(dp_id)
```

Comment:
you rely on the implicit assumption that dp_id will be unique throughout the servers life. This might be true now, but might change in the future. Better adopt a more defensive strategy here.

Reply:
fair comment, this could instead be stored using the clk's upload token. 

With our default config this entire `encodings.py` cache isn't actually used so I'd prefer a good offense as the best defense and remove the whole thing.